client/state/selectors/is-regenerating-jetpack-post-by-email.js

```diff
@@ -15,5 +20,9 @@ import { get } from 'lodash';
  * @return {?Boolean}            Whether Post by Email address is currently being updated
  */
 export default function isRegeneratingJetpackPostByEmail( state, siteId ) {
-	return get( state.jetpack.settings.requests, [ siteId, 'regeneratingPostByEmail' ], null );
+	return get(
+		getRequest( state, saveJetpackSettings( siteId, { post_by_email_address: 'regenerate' } ) ),
```

Comment:
If we're touching this selector to rely on the data layer, and we don't update the UI to use the new action, it'll break the UI a bit by not properly showing a loading indicator. However, if we want to use the new `regeneratePostByEmail` action creator in this PR, we'll lose the automatic updating of the email once regenerated, which is even worse 😞 

We should probably act quickly and continue switching the whole thing to the data layer shortly after we merge this one. Another option is to keep the `isRegenerating` selector changes for another PR.